trending_deploy/deploy.py

```diff
@@ -1,32 +1,95 @@
-
-from typing import List, Literal
-
-from trending_deploy.constants import Model
+from typing import List, Literal, Dict, Any
+import os
+from huggingface_hub import (
+    InferenceClient,
+    create_inference_endpoint, 
+    get_inference_endpoint,
+    list_inference_endpoints,
+    model_info as get_model_info
+)
+
+from trending_deploy.constants import Model, MEMORY_USAGE_TO_INSTANCE, Instance, INSTANCES
+from trending_deploy.models import get_num_parameters_from_model, get_viable_instance_from_num_parameters
+
+# Configuration
+HF_TOKEN = os.environ.get("HF_TOKEN")
+VENDOR = "aws"
+REGION = "us-east-1"
+TYPE = "open"
+NAMESPACE = "hf-inference" 
+ENDPOINT_PREFIX = "trending-deploy-"
+
+# Instance size mapping based on instance memory
+# Maps instance memory to HF instance size (x1, x2, etc.)
+INSTANCE_SIZE_MAPPING = {
+    instance.memory_usage_bytes: f"x{2**(i)}" 
+    for i, instance in enumerate(sorted(INSTANCES, key=lambda x: x.memory_usage_bytes))
+}
 
 
 def load_deployed_models() -> List[str]:
     """
     Load the list of models that are already deployed.
     """
-    # TODO
-    return []
-
-
-def deploy_model(model_name: str) -> bool:
+    try:
+        endpoints = list_inference_endpoints(namespace=NAMESPACE)
+        # Extract model names from endpoints starting with our prefix
+        deployed_models = []
+        for endpoint in endpoints:
+            if endpoint.name.startswith(ENDPOINT_PREFIX):
+                # Extract the model name from the repository field
+                deployed_models.append(endpoint.repository)
+        return deployed_models
+    except Exception as e:
+        print(f"Error loading deployed models: {e}")
+        return []
+
+
+def deploy_model(model: Model) -> bool:
     """
     Deploy the specified model.
 
     Args:
-        model_name (str): The name of the model to deploy.
+        model (Model): The Model object containing model_info and viable_instance.
 
     Returns:
         bool: True if the model was successfully deployed, False otherwise.
     """
-    # TODO
-    return False
-
-
-def undeploy_model(model_name: str):
+    try:
+        model_name = model.model_info.id
+        endpoint_name = f"{ENDPOINT_PREFIX}{model_name.replace('/', '-')}"
+        
+        # Get task from model info
+        task = model.model_info.pipeline_tag or "text-generation"
+        
+        # Get instance size directly from the viable_instance memory
+        instance_size = INSTANCE_SIZE_MAPPING.get(
+            model.viable_instance.memory_usage_bytes, 
+            "x1"  # Default to x1 if mapping not found
+        )
+
+        endpoint = create_inference_endpoint(
+            name=endpoint_name,
+            repository=model_name,
+            framework="pytorch",
+            task=task,
+            accelerator="cpu",
+            vendor=VENDOR,
+            region=REGION,
+            type=TYPE,
+            instance_size=instance_size,
+            instance_type="intel-icl"
```

Comment:
Is this the only option? I'm not very familiar with it. I do see "intel-icl" as an option in the [docs](https://huggingface.co/docs/huggingface_hub/v0.30.2/en/package_reference/hf_api#huggingface_hub.HfApi.create_inference_endpoint).

Reply:
yes it's the way to go